commit-based-cipm/fi-tests/cipm.consistency.fitests.repositorytests/src/cipm/consistency/fitests/repositorytests/util/difffilter/DiffFilter.java

```diff
@@ -0,0 +1,193 @@
+import java.util.regex.Pattern;
+
+/**
+ * A class that can filter out the non-script parts of a given diff patch.
+ * 
+ * @author Alp Torac Genc
+ */
+public class DiffFilter {
+	/**
+	 * Pattern for the beginning of a diff script line.
+	 */
+	private static final Pattern diffLineSignPattern = Pattern.compile("^(?:\\+|-)");
+	/**
+	 * Pattern for a diff script line, which only contains whitespace characters,
+	 * i.e. an empty diff script line.
+	 */
+	private static final Pattern diffLineWhitespacePattern = Pattern
+			.compile(String.format("%s?\\s*$", diffLineSignPattern.pattern()));
+
+	/**
+	 * Pattern for a diff header, i.e. the line that contains the diff command.
+	 */
+	private static final Pattern diffHeaderPattern = Pattern.compile("^\\s*diff --git .*");
+	/**
+	 * Pattern for a hunk header in a diff patch, i.e. the part where a hunk is
+	 * located.
+	 */
+	private static final Pattern diffHunkHeaderPattern = Pattern.compile("^@@ .* @@$");
+
+	/**
+	 * Pattern for file metadata pattern in a diff patch.
+	 */
+	private static final Pattern diffFileMetadataPattern = Pattern.compile("^index .*");
+
+	/**
+	 * Pattern for mode lines in a diff patch.
+	 */
+	private static final Pattern diffModePattern = Pattern.compile("^(?:old|new|new file|deleted file) mode .*");
+	/**
+	 * Pattern for rename lines in a diff patch.
+	 */
+	private static final Pattern diffRenamePattern = Pattern.compile("^rename (?:to|from) .*\\.\\w*");
+	/**
+	 * Pattern for copy lines in a diff patch.
+	 */
+	private static final Pattern diffCopyPattern = Pattern.compile("^copy (?:to|from) .*\\.\\w*");
+
+	/**
+	 * Pattern for a new file state line in a diff patch.
+	 */
+	private static final Pattern diffNewFileStatePattern = Pattern.compile("^\\+\\+\\+ (?:.*\\.\\w*|/.*)");
+	/**
+	 * Pattern for an old file state line in a diff patch.
+	 */
+	private static final Pattern diffOldFileStatePattern = Pattern.compile("^--- (?:.*\\.\\w*|/.*)");
+
+	/**
+	 * Pattern for similarity index line in a diff patch.
+	 */
+	private static final Pattern diffSimilarityIndexPattern = Pattern.compile("^similarity index \\d*\\.?\\d+%");
+	/**
+	 * Pattern for dissimilarity index line in a diff patch.
+	 */
+	private static final Pattern diffDissimilarityIndexPattern = Pattern.compile("^dissimilarity index \\d*\\.?\\d+%");
+
+	/**
+	 * The line in a diff patch that indicates the absence of a newline character at
+	 * the end of a file.
+	 */
+	private static final String diffNoNewLineMessage = "\\ No newline at end of file";
+
+	/**
+	 * The line separator used by the current OS.
+	 */
+	private static final String lineSeparator = System.lineSeparator();
+
+	/**
+	 * @return Splits the given (multi-line) text into its lines, where lines are
+	 *         separated via the line separator used by the system.
+	 */
+	public List<String> splitLines(String text) {
+		var lines = new ArrayList<String>();
+
+		// Do not use System.lineSeparator since GIT uses UNIX terminal
```

Comment:
This comment contradicts the value of `lineSeparator`, which contains exactly `System.lineSeparator`. For best compatibility, `lineSeparator` should equal the regex `\r?\n`.